base/reduce.jl

```diff
@@ -610,62 +645,108 @@ function mapreduce_impl(f, op::Union{typeof(max), typeof(min)},
 end
 
 """
-    maximum(f, itr)
+    maximum(f, itr; [init])
 
 Returns the largest result of calling function `f` on each element of `itr`.
 
+If provided, `init` must be a neutral element for `max` (i.e. which is less
+than or equal to any other element) that will be returned for empty collections.
+
+!!! compat "Julia 1.6"
+    Keyword argument `init` requires Julia 1.6 or later.
+
 # Examples
 ```jldoctest
 julia> maximum(length, ["Julion", "Julia", "Jule"])
 6
+
+julia> maximum(length, []; init=-1)
+-1
```

Comment:
```suggestion
julia> maximum(length, []; init=-Inf)
-Inf
```

Reply:
I actually think `init=-1` here is a good illustrative example.  Since you know that `length(thing)` cannot be less than `0`, `init=-1` is a good neutral element.  So, I suggest to keep it this way.

I also changed the `minimum` demo to `minimum(length, []; init=typemax(Int64))` based on @timholy's suggestion.  Maybe this version clarifies the usage more.  I also added `minimum(tanh, Real[]; init=1.0)` and `maximum(tanh, Real[]; init=-1.0)` just in case it's useful.

Reply:
Fine with me. (Though I must admit I don't remember offhand how `tanh` behaves...)